src/mqtt.c

```diff
@@ -499,7 +505,22 @@ static int mqtt_write(const data_set_t *ds, const value_list_t *vl,
     return status;
   }
 
+  if (conf->pattern != NULL) {
+    if (regexec(&conf->topic_pattern, topic, 0, NULL, 0) == REG_NOMATCH) {
+      return 0;
+    }
+  }
+
   status = format_values(payload, sizeof(payload), ds, vl, conf->store_rates);
+  if (!conf->store_timestamp) {
+    char *p = strchr(payload, ':');
+    int len = strlen(p);
+    if (len > 0) {
+      p = p + 1;
+    }
+    strcpy(payload, p);
```

Comment:
`payload` and `p` overlap, which will cause `strcpy` to sefault. Use `memmove()` instead (it explicitly supports overlapping regions of memory).